### ISSUE TITLE
Changed behaviour in handling template variables passed via cli

### ISSUE THREAD

**Issue by jamesgrimmett** [cylc/cylc-flow#6058]
### Description

I think that there has been a change in behaviour in handling template variables passed to workflows via the `-S` cli option in `v8.2.5`.
In `v8.2.3`, a template variable passed via `-S` would override that variable set in optional config files. In `v8.2.5`, it looks like this behaviour is reversed, and the setting in optional config will override the setting passed to `-S`.

### Reproducible Example

Example test suite containing;
```
test_workflow
├── flow.cylc
├── rose-suite.conf
├── opt
    └── rose-suite-common.conf
```
[flow.cylc]
```
#!jinja2
[scheduler]
   allow implicit tasks=True

[scheduling]
  initial cycle point = {{ START_CYCLE }}
  final cycle point = {{ END_CYCLE }}
  [[ graph]]
    R1 = cold_start
    T00 = """
      cold_start => a
    a => b
    """
```
[opt/rose-suite-common.conf]
```
[template variables]
START_CYCLE="20240101T00"
END_CYCLE="20240102T00"
```

Variables passed using `-S` are ignored, and the workflow runs with the settings from `opt/rose-suite-common.conf`;
```
$ cylc --version
8.2.5

$ cylc vip -O common -S 'START_CYCLE="20230102"' -S 'END_CYCLE="20230102"' .

$ ls ~/cylc-run/test_workflow/run1/log/job/
20240101T0000Z  20240102T0000Z
```

### Expected Behaviour
The setting passed via `-S` overrides other config
```
$ cylc --version
8.2.3

$ cylc vip -O common -S 'START_CYCLE="20230102"' -S 'END_CYCLE="20230102"' .

$ ls ~/cylc-run/test_workflow/run1/log/job/
20230102T0000Z
```

I seems that I get the behaviour that I'm looking for if I use `-s` instead of `-S`, so I could potentially use that as a workaround, but I don't have a good understanding of the difference between these two options. I note that `-s` will currently not allow variables to be updated on reload (#5965 / cylc/cylc-flow#6045), are there any other significant differences between these two options (i.e., `-s` vs `-S`)?

```
$ cylc --version
8.2.5

$ cylc vip -O common -s 'START_CYCLE="20230102"' -s 'END_CYCLE="20230102"' .

$ ls ~/cylc-run/test_workflow/run1/log/job/
20230102T0000Z
```

**Comment by wxtim** [comment]
## Overview of solution

### For Cylc 8.2.x:

#6059  - Cleaned all 3 config items on this branch.

### For Cylc 8.3.0 and Cylc Rose 1.4.0 (master branches in both repos)

#6060 - remove all the cleaning code from Cylc-flow
https://github.com/cylc/cylc-rose/pull/308 - added the option removal to Cylc Rose.

**Comment by oliver-sanders** [comment]
Closed by #312
